hxsl/Ast.hx

```diff
@@ -380,6 +381,16 @@ class Tools {
 		return false;
 	}
 
+	public static function hasBorrowQualifier( v : TVar, path : String ) {
+		if ( v.qualifiers != null )
+			for( q in v.qualifiers )
+				switch (q) {
+					case Borrow(s): return StringTools.startsWith(path, s);
+					default:

```

Comment:
Why startsWith ? seems like a problem if you  @borrow(MyShader) var x : Int  and you have a MyShaderB with a var x as well, you might borrow the wrong variable

Reply:
A leftover code where I attempted to let borrower to use different name compared to source variable. Will change to strict `==`.